### ISSUE TITLE
fail to decode big non-2007 sections

### ISSUE THREAD

**Issue by rurban** [LibreDWG/libredwg#93]
read_R2004_section_info has a overflow problem, the 2007 variant read_data_section not

e.g. 
```
$ programs/dwgread test/test-big/2004/HARTA_E_PRISHTINES.dwg 
Warning: Oops, Section[7] => Section_Info 2111
Warning: Failed to find section_info[13] with type 0x3
ERROR: Failed to read compressed class section
Warning: Failed to find section_info[13] with type 0x1
Warning: Skip empty section 2098 AcDb:AcDbObjects
Warning: Skip empty section 2099 AcDb:AcDbObjects
Warning: Skip empty section 2100 AcDb:AcDbObjects
Warning: Skip empty section 2101 AcDb:AcDbObjects
Warning: Failed to find section_info[13] with type 0x4
```
They were created by AutoCAD and can be opened with AutoCAD.
This was previously hacked around with Oops2, Section[%d] => Section_Info %d"

The root cause is that the section_size is uint64, not int32. (ODA p28)

**Comment by rurban** [comment]
Note that libdxfrw has the very same problems with the testcases from #132

**Comment by rurban** [comment]
The real problem was that bytes_remaining overflowed to negative, but was unsigned. Fixed with c0163ccc71bcd6755ab1f6929b947bcd9ea67c43
